packages/shortest/src/core/test-generator/index.ts

```diff
@@ -0,0 +1,157 @@
+
+  private async testFileExists(): Promise<boolean> {
+    try {
+      await fs.access(this.outputPath);
+      return true;
+    } catch {
+      return false;
+    }
+  }
+
+  private async generateTestFile(): Promise<void> {
+    const rawFileContent = await this.generateRawFileOutput();
+    const formattedCode = await formatCode(rawFileContent, this.rootDir);
+    const lintedCode = await lintCode(formattedCode, this.rootDir);
+
+    try {
+      await fs.mkdir(SHORTEST_DIR_PATH, { recursive: true });
+      await fs.writeFile(this.outputPath, lintedCode);
+      this.log.info("Test file generated successfully", {
+        path: this.outputPath,
+      });
+    } catch (error) {
+      this.log.error("Failed to write tests to file", getErrorDetails(error));
+      throw error;
+    }
+  }
+
+  private async generateRawFileOutput(): Promise<string> {
+    const testPlans = await this.getTestPlans();
+
+    const importStatement = t.importDeclaration(
+      [
+        t.importSpecifier(
+          t.identifier(SHORTEST_NAME),
+          t.identifier(SHORTEST_NAME),
+        ),
+      ],
+      t.stringLiteral("@antiwork/shortest"),
+    );
+
+    const testStatements = testPlans
+      .map((plan) => {
+        const statements: t.Statement[] = [];
+
+        const statementArgs: any[] = [t.stringLiteral(plan.steps[0])];
+
+        if (plan.options?.requiresAuth) {
+          statementArgs.push(
+            t.objectExpression([
+              t.objectProperty(
+                t.identifier("email"),
+                t.memberExpression(
+                  t.memberExpression(
+                    t.identifier("process"),
+                    t.identifier("env"),
+                  ),
+                  t.identifier("SHORTEST_LOGIN_EMAIL"),
+                ),
+              ),
+              t.objectProperty(
+                t.identifier("password"),
+                t.memberExpression(
+                  t.memberExpression(
+                    t.identifier("process"),
+                    t.identifier("env"),
+                  ),
+                  t.identifier("SHORTEST_LOGIN_PASSWORD"),
+                ),
+              ),
+            ]),
+          );
+        }
+
+        const shortestCall = t.callExpression(
+          t.identifier(SHORTEST_NAME),
+          statementArgs,
+        );
+
+        const expectChain = plan.steps.slice(1).reduce((acc, step) => {
+          const expectCall = t.callExpression(
+            t.memberExpression(acc, t.identifier(SHORTEST_EXPECT_NAME)),
+            [t.stringLiteral(step)],
+          );
+          return expectCall;
+        }, shortestCall);
+
+        statements.push(t.expressionStatement(expectChain));
+        return statements;
+      })
+      .flat();
+
+    const program = t.program([importStatement, ...testStatements]);
+
+    return generate(program, {
+      retainLines: true,
+      compact: false,
+    }).code;
+  }
+
+  private async getTestPlans(): Promise<TestPlan[]> {
```

Comment:
For the MVP, no check has been added to ensure the [plan is the current version, the current branch/commit match](https://github.com/antiwork/shortest/blob/53ea0f914d51f38ddea8f0985ffa364d28151697/packages/shortest/src/core/test-planner/index.ts#L125-L129), etc.